src/nvim/shada.c

```diff
@@ -3937,15 +3944,28 @@ static ShaDaReadResult shada_read_next_item(ShaDaReadDef *const sd_reader,
       entry->data.global_var.name =
           xmemdupz(unpacked.data.via.array.ptr[0].via.bin.ptr,
                    unpacked.data.via.array.ptr[0].via.bin.size);
-      if (msgpack_to_vim(unpacked.data.via.array.ptr[1],
-                         &(entry->data.global_var.value)) == FAIL) {
+      SET_ADDITIONAL_ELEMENTS(unpacked.data.via.array, 2,
```

Comment:
This feels meh.
Also, I need to check that this even works properly.

Reply:
how does it work? will existing shada files still be correctly read to reproduce stored strings?

Reply:
For blobs, I use the `global_var.additional_elements` list to store a dummy `VAR_TYPE_BLOB` with the blob.
(the exact value doesn't matter right now as `global_var.additional_elements` doesn't seem to be used for anything else yet)

This seems to work and existing shadas should be (and appear to be) fine.